### PR TITLE
Add a Makeswift example

### PR DESCRIPTION
**Pull request by apledger**
This adds an example of how to use Makeswift with Next.js under `cms-makeswift/`.

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm lint`
- [x] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: examples/cms-makeswift/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
```
Comment (balazsorban44) [inline examples/cms-makeswift/package.json:7]:
```suggestion
    "start": "next start"
```

File: examples/cms-makeswift/package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  },
+  "dependencies": {
+    "@makeswift/runtime": "0.0.21",
+    "next": "latest",
+    "react": "18.1.0",
+    "react-dom": "18.1.0"
```
Comment (balazsorban44) [inline examples/cms-makeswift/package.json:13]:
```suggestion
    "react": "18.2.0",
    "react-dom": "18.2.0"
```

File: examples/cms-makeswift/.env [hunk 3]
```diff
@@ -0,0 +1,2 @@
+MAKESWIFT_API_HOST=https://api.makeswift.com
```
Comment (balazsorban44) [inline examples/cms-makeswift/.env:1]: Let's rename this file to `.env.local.example` to match other examples and be sure that API_KEY's won't be accidentally committed.

File: examples/cms-makeswift/README.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+   ```bash
+   yarn install
+   # or
+   npm install
+   # or
+   pnpm install
+   ```
+
+3. Update the `.env` file to include your Makeswift site's API key:
```
Comment (balazsorban44) [inline examples/cms-makeswift/README.md:39]:
```suggestion
3. Rename the `.env.local.example` file to `.env.local` and include your Makeswift site's API key:
```

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks, just some small suggestions.

**Comment by apledger** [comment]
Thanks for the review @balazsorban44. Fixed all issues.

File: docs/advanced-features/preview-mode.md [hunk 5]
```diff
@@ -25,6 +25,7 @@ description: Next.js has the preview mode for statically generated pages. You ca
     <li><a href="https://github.com/vercel/next.js/tree/canary/examples/cms-kontent">Kontent Example</a> (<a href="https://next-blog-kontent.vercel.app//">Demo</a>)</li>
     <li><a href="https://github.com/vercel/next.js/tree/canary/examples/cms-umbraco-heartcore">Umbraco Heartcore Example</a> (<a href="https://next-blog-umbraco-heartcore.vercel.app/">Demo</a>)</li>
     <li><a href="https://github.com/vercel/next.js/tree/canary/examples/cms-plasmic">Plasmic Example</a> (<a href="https://nextjs-plasmic-example.vercel.app/">Demo</a>)</li>
+    <li><a href="https://github.com/vercel/next.js/tree/canary/examples/cms-makeswift">Makeswift Example</a> (<a href="https://makeswift-examples-basic-typescript.vercel.app/">Demo</a>)</li>
```
Comment (balazsorban44) [inline docs/advanced-features/preview-mode.md:28]: Not a blocker to merge the PR, but do you want to redeploy this on https://next-blog-makeswift.vercel.app instead for consistency?

File: examples/cms-makeswift/README.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
+# A statically generated landing page with Next.js and Makeswift
+
+This example showcases how you can use [Makeswift](https://www.makeswift.com/) to visually build statically generated pages in Next.js.
+
+## Deploy your own
+
+Deploy the example using [Vercel](https://vercel.com?utm_source=github&utm_medium=readme&utm_campaign=next-example):
+
+[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https://github.com/vercel/next.js/tree/canary/examples/DIRECTORY_NAME&project-name=cms-makeswift&repository-name=cms-makeswift)
```
Comment (balazsorban44) [inline examples/cms-makeswift/README.md:9]:
```suggestion
[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https://github.com/vercel/next.js/tree/canary/examples/cms-makeswift&project-name=cms-makeswift&repository-name=cms-makeswift)
```

File: examples/cms-makeswift/README.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+## Demo
+
+### [https://nextjs-makeswift-example.vercel.app/](https://nextjs-makeswift-example.vercel.app/)
```
Comment (balazsorban44) [inline examples/cms-makeswift/README.md:13]: See also: https://github.com/vercel/next.js/pull/38018/files#r907833127

Up to you where to point, but at least the README and the link in the docs should probably link to the same example. :+1:

**Review by balazsorban44 — DISMISSED** [review]

**Review by leerob — APPROVED** [review]

**Pull request merged**